lib/liquid/context.rb

```diff
@@ -27,7 +27,7 @@ def initialize(environments = {}, outer_scope = {}, registers = {}, rethrow_erro
       @this_stack_used = false
 
       if rethrow_errors
-        self.exception_handler = ->(e) { raise }
+        self.exception_handler = ->(e) { raise e }
```

Comment:
It seems `raise` pickups up the error even when called in a lambda, as long as the lambda is called from within the `rescue` block:

```ruby
h1 = ->(e){ raise }
h2 = ->(e){ raise e }
begin; raise ArgumentError, 'test'; rescue => e; h1.(e); end
begin; raise ArgumentError, 'test'; rescue => e; h2.(e); end
# both still result in => ArgumentError: test
```